pkg/resource/record_set/utils.go

```diff
@@ -0,0 +1,18 @@
+package record_set
+
+import (
+	"strings"
+)
+
+// decodeRecordName filters the DNSName of a record set. ListResourceRecordSets returns
+// the DNS name with a "." at the end and encodes the value for "*", so the DNSName needs
+// to be parsed before comparing with our spec values.
+func decodeRecordName(name string, specName string) string {
+	if specName[len(specName)-1:] != "." {
+		name = name[:len(name)-1]
+	}
+	if strings.Contains(name, "\\052") {
+		return strings.Replace(name, "\\052", "*", -1)
+	}
+	return name
+}
```

Comment:
While testing simple and alias records, I've noticed ListResourceRecordSets seems to always append a `.` to the end of a DNS name even if we've supplied a CR that does not have the trailing `.` (not the case for CNAME values though). Below is an example CR

```
apiVersion: route53.services.k8s.aws/v1alpha1
kind: RecordSet
metadata:
  name: alias-record
spec:
  name: "*.some.random.example.ack.com"
  hostedZoneID: <hosted-zone-id>
  recordType: A
  aliasTarget:
    evaluateTargetHealth: false
    dnsName: vpce-something-random.com
    hostedZoneID: <vpc-endpoint-hosted-zone-id>
```

In the above case for `spec.name`, the result that we get from `ListResourceRecordSets` would be `\\052.some.random.example.ack.com.`